odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/repository/reactive/ReactiveAbstractCRUDRepository.java

```diff
@@ -0,0 +1,254 @@
+        return jooqReactiveOperations
+            .mono(DSL.deleteFrom(recordTable).where(getCondition(id)).returning())
+            .map(this::recordToPojo);
+    }
+
+    @Override
+    public Flux<P> delete(final Collection<Long> ids) {
+        return jooqReactiveOperations
+            .flux(DSL.deleteFrom(recordTable).where(getCondition(ids)).returning())
+            .map(this::recordToPojo);
+    }
+
+    protected Mono<R> insertOne(final R record) {
+        return jooqReactiveOperations.mono(DSL.insertInto(recordTable).set(record).returning());
+    }
+
+    protected Mono<R> updateOne(final R record) {
+        if (updatedAtField != null) {
+            record.set(updatedAtField, LocalDateTime.now());
+        }
+
+        return jooqReactiveOperations
+            .mono(DSL.update(recordTable).set(record).where(idField.eq(record.get(idField))).returning());
+    }
+
+    protected Flux<R> insertMany(final List<R> records) {
+        InsertSetStep<R> insertStep = DSL.insertInto(recordTable);
+
+        for (int i = 0; i < records.size() - 1; i++) {
+            insertStep = insertStep.set(records.get(i)).newRecord();
+        }
+
+        return jooqReactiveOperations
+            .flux(insertStep.set(records.get(records.size() - 1)).returning(recordTable.fields()));
+    }
+
+    @ReactiveTransactional
+    protected Flux<R> updateMany(final List<R> records) {
+        // TODO: update
+        return Flux.fromIterable(records).flatMap(this::updateOne);
+    }
+
+    protected Mono<Long> fetchCount(final String nameQuery) {
+        return Mono.defer(() -> jooqReactiveOperations
+            .mono(DSL.selectCount().from(recordTable).where(listCondition(nameQuery)))
+            .map(r -> r.component1().longValue()));
+    }
+
+    protected List<Condition> listCondition(final String nameQuery) {
+        return StringUtils.isNotEmpty(nameQuery) ?
+            List.of(nameField.containsIgnoreCase(nameQuery))
+            : emptyList();
+    }
+
+    protected List<Condition> getCondition(final long id) {
+        return List.of(idField.eq(id));
+    }
+
+    protected List<Condition> getCondition(final Collection<Long> ids) {
+        return List.of(idField.in(ids));
+    }
+
+    protected SelectConditionStep<R> baseSelectManyQuery(final String query) {
+        return DSL
+            .selectFrom(recordTable)
+            .where(listCondition(query));
+    }
+
+    private R createRecord(final P pojo, final LocalDateTime updatedAt) {
+        final R record = pojoToRecord(pojo);
+
+        if (updatedAtField != null) {
+            record.set(updatedAtField, updatedAt);
+        }
+
+        return record;
+    }
+
+    protected R pojoToRecord(final P pojo) {
+        return jooqReactiveOperations.newRecord(recordTable, pojo);
+    }
+
+    protected P recordToPojo(final R r) {
+        return r.into(recordTable).into(pojoClass);
+    }
+
+    protected Select<? extends Record> paginate(final Select<R> baseSelect,
```

Comment:
may be add overloaded paginate method with predefined orderField Id and sort order ASC?